enterprise/server/mockoauth/mockoauth.go

```diff
@@ -0,0 +1,253 @@
+package mockoauth
```

Comment:
I think we should call this something besides mockauth -- since mocks are usually associated with test-only code and this is clearly not that, it's confusing for other folks working on the code, but also for customers in the config.

Instead of mockauth, what do you think about singleuserauth or selfauth?

Reply:
+1 for selfauth, a little more concise while still communicating the purpose well